[2.1-develop] Add customer login url from Customer Url model to checkout config so …

quisse opened the pull request:
…it contains the referer url if necessary

…it contains the referer url if necessary

### Description
Use the url created in the customer model instead of creating a new one with the url builder. The current way isn't adding a referrer parameter when required

### Fixed Issues (if relevant)
<!--- Provide a list of fixed issues in the format magento/magento2#<issue_number>, if relevant  -->
1. magento/magento2#12627: Referer is not added to login url in checkout config

### Manual testing scenarios
<!--- Provide a set of unambiguous steps to test the proposed code change -->
1. Enable `Redirect Customer to Account Dashboard after Logging in` in the Magento backend.
2. Retrieve the login url from the `window.checkout` object (key is customerLoginUrl)

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

Commit:
Add customer login url from Customer Url model to checkout config so it contains the referer url if necessary

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=12629) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=12629) before we can accept your contribution.<br/>

Commit:
fixed unit test for checkout configprovider

# Conflicts:
#	app/code/Magento/Customer/Test/Unit/Model/Checkout/ConfigProviderTest.php

Commit:
removed not yet existing method call

dmanners commented:
Backport for https://github.com/magento/magento2/pull/12630

omiroshnichenko commented:
Please, review comments in https://github.com/magento/magento2/pull/12631

Commit:
updated code following code review & backward compatible development guide

# Conflicts:
#	app/code/Magento/Customer/Test/Unit/Model/Checkout/ConfigProviderTest.php

Commit:
updated code following code review & backward compatible development guide

Commit:
fixed code styles

Commit:
fixed older phpunit version

Commit:
add nullable in comment & code cleanup

Commit:
code cleanup

Commit:
replaced null coalescing operator because of php 5.6

omiroshnichenko commented on app/code/Magento/Customer/Model/Checkout/ConfigProvider.php:
```diff
@@ -30,19 +32,28 @@ class ConfigProvider implements ConfigProviderInterface
         $this->urlBuilder = $urlBuilder;
         $this->storeManager = $storeManager;
         $this->scopeConfig = $scopeConfig;
+        $this->customerUrl = $customerUrl?: ObjectManager::getInstance()
```
Add space before question mark.

Commit:
code cleanup

omiroshnichenko committed:
Merge branch '2.1-develop' of github.com:magento/magento2 into quisse-2.1
